xbmc/utils/QueueAndPlayUtils.cpp

```diff
@@ -0,0 +1,339 @@
+/*
+ *  Copyright (C) 2016-2022 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#include "Application.h"
+#include "GUIPassword.h"
+#include "PartyModeManager.h"
+#include "PlayListPlayer.h"
+#include "QueueAndPlayUtils.h"
+#include "ServiceBroker.h"
+#include "dialogs/GUIDialogBusy.h"
+#include "filesystem/File.h"
+#include "guilib/GUIComponent.h"
+#include "guilib/GUIWindowManager.h"
+#include "playlists/PlayList.h"
+#include "playlists/PlayListFactory.h"
+#include "profiles/ProfileManager.h"
+#include "settings/SettingsComponent.h"
+#include "utils/log.h"
+#include "view/GUIViewState.h"
+
+using namespace XFILE;
+using namespace std::chrono_literals;
+
+// set and reset in worker thread, read by calling thread
+CEvent CQueueAndPlayUtils::m_utilsJobEvent(false, false);
+
+class CAddItemToPlaylistJob : public CJob
+{
+  const CFileItemPtr m_pItem;
+  CFileItemList& m_queuedItems;
+
+public:
+  CAddItemToPlaylistJob(const CFileItemPtr& pItem, CFileItemList& queuedItems)
+    : m_pItem(pItem), m_queuedItems(queuedItems)
+  {
+  }
+
+  ~CAddItemToPlaylistJob(void) override = default;
+
+  bool DoWork(void) override
+  {
+    CQueueAndPlayUtils queueAndPlayUtils;
+    queueAndPlayUtils.JobStart();
+    queueAndPlayUtils.AddToPlayList(m_pItem, m_queuedItems);
+    queueAndPlayUtils.JobFinish(); // Signal job finished
+    return true;
+  }
+};
+
+bool CQueueAndPlayUtils::PlayItem(const CFileItemPtr& itemToPlay,
+                                  const bool bShowBusyDialog /* = true */)
```

Comment:
please don't pass a plain const bool and avoid new default arguments

```suggestion
bool CQueueAndPlayUtils::PlayItem(const CFileItemPtr& itemToPlay,
                                  bool bShowBusyDialog)
```

Reply:
Why not?

Reply:
passing plain const by value to (or returning from) a function has no benefit and may interfere with move semantics. not in this case, but that is why we generally don‘t want to do it.

default arguments are an anti-pattern, they obscure from the calling side.
so if there‘s a chance to kill it -> then kill it 🙂

Reply:
OK, makes sense.  I just followed what had gone before really, but I can easily kill it.  TY for your input :)